src/backend/renderer/output/element/mod.rs

```diff
@@ -0,0 +1,443 @@
+use crate::{
+    backend::renderer::{ImportAll, Renderer},
+    utils::{Physical, Point, Rectangle, Scale},
+};
+
+mod id;
+
+pub use id::Id;
+use wayland_server::protocol::wl_buffer;
+
+pub mod surface;
+pub mod texture;
+
+/// The underlying storage for a element
+#[derive(Debug)]
+pub enum UnderlyingStorage<'a, R: Renderer> {
+    /// A wayland buffer
+    Wayland(wl_buffer::WlBuffer),
+    /// A texture
+    External(&'a R::TextureId),
+}
+
+/// A single render element
+pub trait RenderElement<R: Renderer> {
+    /// Get the unique id of this element
+    fn id(&self) -> &Id;
+    /// Get the current commit position of this element
+    fn current_commit(&self) -> usize;
+    /// Get the location relative to the output
+    fn location(&self, scale: Scale<f64>) -> Point<i32, Physical>;
```

Comment:
Given both location and geometry are using Physical coordinates now, could this have a default impl, that just calls `self.geometry().loc`?

Reply:
:+1: done